lib/battle/AccessibilityInfo.h

```diff
@@ -27,15 +27,15 @@ enum class EAccessibility
 	DESTRUCTIBLE_WALL,
 	GATE, //sieges -> gate opens only for defender stacks
 	UNAVAILABLE, //indestructible wall parts, special battlefields (like boat-to-boat)
-	SIDE_COLUMN //used for first and last columns of hexes that are unavailable but wat machines can stand there
+	SIDE_COLUMN //used for first and last columns of hexes that are unavailable but war machines can stand there
 };
 
 
 using TAccessibilityArray = std::array<EAccessibility, GameConstants::BFIELD_SIZE>;
 
 struct DLL_LINKAGE AccessibilityInfo : TAccessibilityArray
 {
-	std::map<BattleHex, ui8> destructibleEnemyTurns;
+	const std::array<int8_t, GameConstants::BFIELD_SIZE> * destructibleEnemyTurns = nullptr;
```

Comment:
Another reminder that this may break checks on invalid hexes, e.g. towers.

Also - why is this a pointer now?

Reply:
`destructibleEnemyTurns` in `AccessibilityInfo` is just a view on `destructibleEnemyTurns` from `ReachabilityInfo::Parameters`
```cpp
ReachabilityInfo CBattleInfoCallback::getReachability(const ReachabilityInfo::Parameters & params) const
{
	if(params.flying)
		return getFlyingReachability(params);
	else
	{
		auto accessibility = getAccessibility(params.knownAccessible);

		accessibility.destructibleEnemyTurns = & params.destructibleEnemyTurns;

		return makeBFS(accessibility, params);
	}
}
```